src/core/Akka/Event/Logging.cs

```diff
@@ -105,14 +105,12 @@ public static string FromActor(IActorContext actor, ActorSystem system)
 
         public static string FromActorRef(IActorRef a, ActorSystem system)
         {
-            try
-            {
-                return a.Path.ToStringWithAddress(system.AsInstanceOf<ExtendedActorSystem>().Provider.DefaultAddress);
-            }
-            catch // can fail if the ActorSystem (remoting) is not completely started yet
-            {
-                return a.Path.ToString();
-            }
+            Address address = null;
+            if (system is ExtendedActorSystem extendedSystem && !(extendedSystem.Provider is null))
+                address = extendedSystem.Provider.DefaultAddress;
+
+            // maybe the ActorSystem (remoting) is not completely started yet
+            return address is null ? a.Path.ToString() : a.Path.ToStringWithAddress(address);
```

Comment:
This is a good change too

Reply:
There should never be a expected throw (if possible)
This is why `int.TryParse("aa", out var i)` got added,
`try { i = int.Parse("aa"); } catch(FormatException ex) { ... }`  was a perf killer